server/utils/users.js

```diff
@@ -61,19 +57,49 @@ const isValidDob = (dateParam) => {
 		!Number.isFinite(month) ||
 		!Number.isFinite(day)
 	)
-		throw badRequestErr('Invalid DOB');
+		throw badRequestErr(`Invalid ${varName}`);
 	const momentDate = moment(
 		`${year.toString().padStart(4, '0')}-${month
 			.toString()
 			.padStart(2, '0')}-${day.toString().padStart(2, '0')}`
 	);
+	if (!momentDate.isValid()) throw badRequestErr(`Invalid ${varName}`);
+	return momentDate;
+};
+
+/**
+ *
+ * @param {string} date in format MM-DD-YYYY
+ * @returns {string} date string if it is valid otherwise throws an error
+ */
+const isValidDob = (dateParam) => {
+	const momentDate = isValidDateStr(dateParam, 'DOB');
 	if (!momentDate.isValid()) throw badRequestErr('Invalid DOB');
 	const difference = moment().diff(momentDate, 'year');
 	if (difference < 12 || difference > 100)
 		throw badRequestErr('Invalid DOB: should be between 12-100 years in age');
 	return momentDate.format('MM-DD-YYYY');
 };
 
+const isValidFromAndToDate = (fromDate, toDate = null) => {
+	const fromMomentDate = isValidDateStr(fromDate, 'From Date');
+	const toMomentDate = toDate ? isValidDateStr(toDate, 'To Date') : null;
+	if (!fromMomentDate.isValid()) throw badRequestErr('Invalid From Date');
+	if (toMomentDate && !toMomentDate.isValid())
+		throw badRequestErr('Invalid To Date');
+	if (moment().diff(fromMomentDate, 'days') < 0)
+		throw badRequestErr('From Date cannot be in the future');
+	if (toMomentDate) {
+		const difference = toMomentDate.diff(fromMomentDate, 'days');
+		if (difference < 1)
+			throw badRequestErr('To Date cannot be same as or before From Date');
+	}
+	return {
+		from: fromMomentDate.format('MM-DD-YYYY'),
+		to: toMomentDate ? toMomentDate.format('MM-DD-YYYY') : null,
+	};
```

Comment:
How will you be keeping track of your current education? 'to' should either be present or ongoing.

Reply:
@sabithanazareth the to date can either be a date after the from date or null
Null represents that it's on going